### PR TITLE
Conditional batch version for itests

### PR DESCRIPTION
**Pull request by guggero**
Makes the batch version configurable to enable integration testing of different client batch versions in the same batch.

#### Pull Request Checklist
- [ ] LndServices minimum version has been updated if new lnd apis/fields are
  used.

### CODE REVIEW

File: order/batch_verifier.go [hunk 1]
```diff
@@ -75,13 +75,12 @@ type batchVerifier struct {
...
-	// than we use for creating the batch. Otherwise we bail out of the batch.
-	// This should already be handled when the client connects/authenticates.
-	// But doesn't hurt to check again.
+	// First, make sure the server used the same batch version than we use
+	// for creating the batch. Otherwise, we bail out of the batch.
+	// This should already be handled when the client connects/
+	// authenticates. But doesn't hurt to check again.
 	if batch.Version != v.version {
-		return ErrVersionMismatch
+		return NewErrVersionMismatch(v.version)
```
Comment (positiveblue) [inline order/batch_verifier.go:83]: maybe it's worth to add both versions to the log message. Something like

```
client version (%d) mismatches server version (%d)
```

**Review by positiveblue — APPROVED** [review]
LGTM, this will make testing/debugging easier.

File: order/batch_verifier.go (hunk 1 above)
Comment (guggero) [reply order/batch_verifier.go]: Good idea, done.

File: config.go [hunk 2]
```diff
@@ -160,6 +161,15 @@ type Config struct {
...
 	// dialing the auctioneer server.
 	AuctioneerDialOpts []grpc.DialOption
+
+	// DebugConfig is a set of debug options used for development and
+	// testing only.
+	DebugConfig *DebugConfig `group:"debug" namespace:"debug" hidden:"true"`
+}
+
+// DebugConfig is a set of debug options used for development and testing only.
+type DebugConfig struct {
```
Comment (bhandras) [inline config.go:171]: Very nice!

**Review by bhandras — APPROVED** [review]
LGTM :1st_place_medal:

**Pull request merged**